ethcore/builtin/src/lib.rs

```diff
@@ -548,14 +572,39 @@ impl Bn128Pairing {
 	}
 }
 
+impl Implementation for EIP1962 {
+	/// Can fail in many cases, so leave it for an implementation
+	fn execute(&self, input: &[u8], output: &mut BytesRef) -> Result<(), &'static str> {
+		let result = eth_pairings::public_interface::API::run(&input);
```

Comment:
`eth_pairings::public_interface::API` is an odd sequence of names, why not just `eth_pairings::run`? I mean, if it's callable from the outside it's part of the api and part of the public interface so it seems tautological to repeat "api" and "public interface".